tests/template_tests/filter_tests/test_last.py

```diff
@@ -15,3 +15,8 @@ def test_last01(self):
     def test_last02(self):
         output = self.engine.render_to_string('last02', {"a": ["x", "a&b"], "b": ["x", mark_safe("a&b")]})
         self.assertEqual(output, "a&b a&b")
+
+    @setup({'last03': '{% autoescape off %}{{ a|last }}{% endautoescape %}'})
+    def test_last03(self):
```

Comment:
More descriptive test names would be welcome. The existing names are merely from a big test refactoring.